### PR TITLE
Test for portfolios and auth

### PR DESCRIPTION
**Pull request by w89652**
1. Add additional information (amount, portfolio_type, positions) that users must provide for creating portfolios.
2. Ensure that all functions except for forget_password (which still has unresolved bugs Orz) have passed testing.
3. Address some warnings in the codebase.

### CODE REVIEW

File: migrations/2024-02-27-030730_create_schemas/up.sql [hunk 1]
```diff
@@ -123,7 +123,7 @@ CREATE TABLE IF NOT EXISTS portfolio_balance (
     id SERIAL PRIMARY KEY ,
     portfolio_id SERIAL REFERENCES portfolios(id) NOT NULL,
     quantity BIGINT NOT NULL DEFAULT 0,
-    currency_id INTEGER REFERENCES currencies(id) NOT NULL
+    currency_id INTEGER NOT NULL DEFAULT 1 REFERENCES currencies(id)
```
Comment (grorge123) [inline migrations/2024-02-27-030730_create_schemas/up.sql]: Why should it need a default value? If insertion does not provide value, it should throw an error.

File: src/auth/signup.rs [hunk 2]
```diff
@@ -21,7 +21,7 @@ pub(crate) struct SignupInfo<'r> {
 // TODO, signup is available only when not logged in
 // if signup sucessfully, redirect to login page. (It won't log in automatically)
 // Otherwise, return Status::BadRequest and a string indicating the error. (It is not fancy at all :< )
-#[post("/api/auth/user", data = "<signup_info>")]
+#[post("/api/auth/signup", data = "<signup_info>")]
```
Comment (grorge123) [inline src/auth/signup.rs]: The original path is correct. Please refer to [api](https://app.swaggerhub.com/apis-docs/CHODEHIRGD157842/Rudrist/1.0.0#/admin/post_api_auth_user).

File: test/signup.py [hunk 3]
```diff
@@ -10,6 +10,6 @@
   "email": "admin@localhost",
 }
 
-res = session_requests.post(config.url + 'auth/user', json=data)
+res = session_requests.post(config.url + 'auth/singup', json=data)
```
Comment (grorge123) [inline test/signup.py]: Please refer [api](https://app.swaggerhub.com/apis-docs/CHODEHIRGD157842/Rudrist/1.0.0#/admin/post_api_auth_user)

Comment (w89652) [reply test/signup.py]: OK, I have fixed 3 errors above.

**Review by grorge123 — APPROVED** [review]

**Pull request merged**